src/main/java/edu/cornell/kfs/fp/service/impl/AmazonWebServicesBillingServiceImpl.java

```diff
@@ -0,0 +1,649 @@
+            transactionDTO.setFailingError(true);
+        }
+    }
+    
+    private Account determineWhichAccountToUse(String costCenter, String kfsAccountString) {
+        Account costAccount = buildAccount(costCenter);
+        if (validAccount(costAccount)) {
+            return costAccount;
+        }
+        Account kfsAccount = buildAccount(kfsAccountString);
+        if (validAccount(kfsAccount)) {
+            return kfsAccount;
+        }
+        return null;
+    }
+    
+    private void updateTransactionDTOChartAccountObjectCodes(TransactionDTO transactionDTO, Account account) {
+        transactionDTO.setAccountNumber(account.getAccountNumber());
+        transactionDTO.setChartCode(account.getChartOfAccountsCode());
+        transactionDTO.setObjectCodeNumber(getTransactionObjectCode());
+    }
+    
+    private void updateTransactionDTOObjectCodeNumbers(TransactionDTO transactionDTO, String costCenterObjectCode,
+            String costCenterSubObjectCode) {
+        if (StringUtils.isNotBlank(costCenterObjectCode)) {
+            ObjectCode objectCode = getObjectCodeService().getByPrimaryIdForCurrentYear(transactionDTO.getChartCode(), costCenterObjectCode);
+            if (objectCode != null) {
+                transactionDTO.setObjectCodeNumber(objectCode.getFinancialObjectCode());
+                if (StringUtils.isNotBlank(costCenterSubObjectCode)) {
+                    SubObjectCode subObjectCode = getSubObjectCodeService().getByPrimaryIdForCurrentYear(transactionDTO.getChartCode(), 
+                            transactionDTO.getAccountNumber(), objectCode.getFinancialObjectCode(), costCenterSubObjectCode);
+                    if (subObjectCode != null) {
+                        transactionDTO.setSubObjectCodeNumber(subObjectCode.getFinancialSubObjectCode());
+                    } else {
+                        transactionDTO.setFailingError(true);
+                    }
+                }
+            } else {
+                transactionDTO.setFailingError(true);
+            }
+        } else {
+            transactionDTO.setObjectCodeNumber(getTransactionObjectCode());
+        }
+        
+    }
+
+    private void updateTransactionDTOProjectCodeNumber(TransactionDTO transactionDTO, String costCenterProjectCode) {
+        if (StringUtils.isNotBlank(costCenterProjectCode)) {
+            ProjectCode projectCode = getProjectCodeService().getByPrimaryId(costCenterProjectCode);
+            if (projectCode != null) {
+                transactionDTO.setProjectCodeNumber(projectCode.getCode());
+            } else {
+                transactionDTO.setFailingError(true);
+            }
+        }
+    }
+    
+    private Account buildAccount(String accountNumber) {
+        if (StringUtils.isNotBlank(accountNumber)) {
+            try {
+                Account account = getAccountService().getUniqueAccountForAccountNumber(accountNumber);
+                return account;
+            } catch (Exception e) {
```

Comment:
Does the method above throw something more specific than Exception? If so, the catch should have that specific exception. If not, bummer.